pkg/unidling/api/types.go

```diff
@@ -9,6 +9,10 @@ const (
 	// objects associated with the idled endpoints
 	UnidleTargetAnnotation = "idling.alpha.openshift.io/unidle-targets"
 
+	// PreviousScaleAnnotation contains the previous scale of a scalable object
+	// (currently only applied by the idler)
+	PreviousScaleAnnotation = "scale.openshift.io/previous-scale"
```

Comment:
> I can, but when I was talking to @smarterclayton, we were discussing the idea of making something like this automatically applied on scales in the future (and then have scale.openshift.io/xyz passed through the scale subresource so that it can act like a prototype API field, just like other objects), so using something like this would future-proof that somewhat.

I'm not really convinced that I'd like to use an annotation to remember previous state (see problems with the other kube attempts to use annotations to store history).  I'm also not seeing a lot of value for "undo" in the general case.  It seems like it would open us up to a new class of privilege escalation attacks if we had a controller that attempted to reconcile this annotation automatically.

I'd rather target the information that we want to preserve for its current usage.

@smarterclayton is there some more complete design that covers how we'd resolve the problems of history tracking annotations and privilege escalation attacks based on controllers watching annotations (think DC scaling existing RC, not scaling the current object)?  Are you ok with scoping this to address our current, narrow need first?